### PR TITLE
Added ansible playbook for adding a jenkins linux slave for openenclave

### PR DESCRIPTION
**Pull request by ahapliuc**
Added ansible playbook for adding a Jenkins linux slave for openenclave.

Currently only works for Linux (Ubuntu16.04 ,might also work for ubuntu 18.04).

Tested the playbook locally with an test instance of Jenkins.

This is related to https://github.com/Microsoft/openenclave/issues/1063

### CODE REVIEW

File: .jenkins/ansible/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+CI Students
```
Comment (ionutbalutoiu) [inline .jenkins/ansible/README.md]: Change the `CI Students` title. I believe this is unrelated to the current PR.

File: .jenkins/ansible/README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+Add the external role:
+
+```
+ansible-galaxy install kobanyan.jenkins-jnlp-slave
+```
+
+Role Variables
+--------------
+
+#
```
Comment (ionutbalutoiu) [inline .jenkins/ansible/README.md]: Document the variables needed.

File: .jenkins/ansible/hosts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+[defaults]
+
+[linux-slaves]
+172.26.179.202
```
Comment (ionutbalutoiu) [inline .jenkins/ansible/hosts]: Remove the actual IP address and add a placeholder

File: .jenkins/ansible/README.md (hunk 2 above)
Comment (ionutbalutoiu) [reply .jenkins/ansible/README.md]: Also, please document that people should edit `hosts` file accordingly.

File: .jenkins/ansible/vars/variables.var [hunk 4]
```diff
@@ -0,0 +1,4 @@
+jenkins_master: "http://172.26.179.203:8080"
```
Comment (ionutbalutoiu) [inline .jenkins/ansible/vars/variables.var]: Remove the actual address and use a a placeholder.

File: .jenkins/ansible/vars/variables.var [hunk 5]
```diff
@@ -0,0 +1,4 @@
...
+jenkins_slave_secret: "eb0a39fdfec932e7847877329a87abe85aa2402fd2f4c6695def9192c9cf360d" #The secret must be taken from when the node is created
```
Comment (ionutbalutoiu) [inline .jenkins/ansible/vars/variables.var]: Same as above. Do not use the actual value, but a placeholder instead.

File: .jenkins/ansible/README.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
+Add Jenkins slave
+=========
+
+This role will install needed Jenkins students (slaves) for both linux and windows. That includes all the needed requirements for openenclave.
+
+Requirements
+------------
+
+All the python requierements are in the requierements.txt and can be installed with:
```
Comment (johnkord) [inline .jenkins/ansible/README.md]: Typo here, should be "requirements" and "requirements.txt"

File: .jenkins/ansible/deploy_jenkins.yml [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+                   - libsgx-enclave-common
+                   - libsgx-enclave-common-dev
+                   - libsgx-dcap-ql 
+                   - libsgx-dcap-ql-dev
+                   - az-dcap-client
+                   - open-enclave 
+         - name: Install jre needed by Jenkins
+           apt: name{{item}} state=present update_cache=yes
+           with_items:
+                   - openjdk-8-jre
```
Comment (johnkord) [inline .jenkins/ansible/deploy_jenkins.yml]: It looks like this needs to install the docker service and add the "jenkins" user to the docker group

Comment (ahapliuc) [reply .jenkins/ansible/deploy_jenkins.yml]: Hello John, this only installs and adds the slave to the Jenkins master, and we do not use docker on the Jenkins slaves AFAIK. Does this make sense, or I missed something and I should add the docker service and Jenkins user on the Jenkins slave?

Comment (ahapliuc) [reply .jenkins/ansible/deploy_jenkins.yml]: @johnkord can you please let me know which is the preferred way?

Comment (ionutbalutoiu) [reply .jenkins/ansible/deploy_jenkins.yml]: @ahapliuc I just checked now. Docker is installed on the Jenkins slaves. Not sure about its purpose, but I think is required. What's the purpose of Docker for Jenkins slaves ? @johnkord

Comment (johnkord) [reply .jenkins/ansible/deploy_jenkins.yml]: @ahapliuc @ionutbalutoiu We use docker in the Jenkinsfile for Open Enclave: https://github.com/Microsoft/openenclave/blob/master/.jenkins/Jenkinsfile

It is used both on the non-ACC VMs and the ACC VMs (though, as of yesterday, the one "non-simulation container" test has been temporarily disabled on the ACC VMs). Take a look at the test script named "simulationTest", which runs its steps inside the "OETOOLS_IMAGE" docker image.

Around half of our tests in CI are run in a container!

Comment (ionutbalutoiu) [reply .jenkins/ansible/deploy_jenkins.yml]: Ohhh, you're absolutely right! I don't know how I forgot about the tests runs. Lol

Comment (ahapliuc) [reply .jenkins/ansible/deploy_jenkins.yml]: Added the docker installation part and also made the user jenkins part of the docker group @ionutbalutoiu @johnkord

Comment (johnkord) [reply .jenkins/ansible/deploy_jenkins.yml]: @ahapliuc I'm looking for the docker changes in the commits and I can't find them!

Comment (ahapliuc) [reply .jenkins/ansible/deploy_jenkins.yml]: @johnkord there should be in place now, committed to a wrong local branch, sorry about that.

**Pull request merged**